src/app/plugins/platform/Docker/plugins/cms/Drupal8/index.ts

```diff
@@ -12,18 +14,20 @@ import {
   enableXdebugProfiler,
   xdebugEnvironment,
 } from '../../../xdebug';
-
-import installDrupal, {
-  drupalProject,
-  pantheonProject,
-  Project,
-} from './installDrupal';
 import createDrupalDockerfile from './createDrupalDockerfile';
 import createDrushDockerfile from './createDrushDockerfile';
-import dedent from 'dedent';
 import { gessoDrupalPath } from '../../gesso/constants';
+import { injectPlatformConfig, renameWebRoot } from './installDrupal';
 import { promptOrUninteractive } from '../../../../../../../util';
 
+const drupalProject = 'drupal-composer/drupal-project:8.x-dev';
+type DrupalProject = typeof drupalProject;
+
+const pantheonProject = 'pantheon-systems/example-drops-8-composer';
+type PantheonProject = typeof pantheonProject;
+
+type Project = PantheonProject | DrupalProject;
```

Comment:
Good opportunity for an enum.

```suggestion
enum Project {
  Drupal = 'drupal-composer/drupal-project:8.x-dev',
  Pantheon = 'pantheon-systems/example-drops-8-composer',
}
```

Reply:
While we're looking at this, I would make two arguments:
1. since we'd be implementing symbolic identifiers, let's give them human-readable names (and avoid keying off of GitHub upstreams/templates directly); and
2. it's probably breaking enough that we should isolate it from this cleanup work.

The enum would look something like this:

```ts
enum ProjectType {
  Drupal = 'drupal',
  Pantheon = 'pantheon',
}
```

We can use the [`Record<Keys, Type>`](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeystype) helper to construct dictionaries from the enum values:

```ts
const projectUpstreams: Record<ProjectType, string> = {
  [ProjectType.Drupal]: 'drupal-composer/drupal-project:8.x-dev',
  [ProjectType.Pantheon]: 'pantheon-systems/example-drops-8-composer',
};
```

I think this abstraction would allow us to change how we scaffold new projects without breaking existing projects, as we can record the project template separate from the project type itself - if a project was already scaffolded, we don't have to run `composer create-project` again.

Reply:
> it's probably breaking enough that we should isolate it from this cleanup work

This is the most notable point that aligns with my concerns introducing it into this work. I do agree that restructuring this piece could be a nice revision, however.

---

I do like the abstraction of project type and upstream you're describing @gustavderdrache and would lean toward doing that. In the spirit of moving toward more consistency with WordPress and using an upstream supporting `composer create-project` I'm also considering refactoring this out to a higher level where the complete project type list might look more like:

```JavaScript
enum ProjectType {
  Drupal8 = 'drupal8',
  D8Pantheon = 'd8Pantheon',
  Drupal9 = 'drupal9',
  D9Pantheon = 'd9Pantheon',
  WordPress = 'wordpress',
  WPPantheon = 'wpPantheon',
}
```

One thing that does stand out here that I'm not sure what to think of is the feeling of redundancy between the keys and values in the way the `ProjectType` enum is listed in either my example or @gustavderdrache's. In both cases, the key/value relationship is mostly just a difference in the casing.

Reply:
You are correct that we don't get much value out of the string... err, values themselves, but we instead get benefits during the edit-, compile-, and run-time phases:

1. When writing code, the TypeScript server can autocomplete `ProjectType` values for us contextually.
2. The compiler has a single source of truth to verify project types in code.
3. At run time, we can use the `ProjectType` object for verification of user input or saved data:

    ```ts
    const projectTypeValues = new Set(Object.values(ProjectType));

    function isProjectTypeValue(value: string): value is ProjectType {
      return projectTypeValues.has(value);
    }
    ````